pygac/klm_reader.py

```diff
@@ -42,6 +47,54 @@
 
 LOG = logging.getLogger(__name__)
 
+
+class KLM_QualityIndicator(IntFlag):
+    """Quality Indicators
+
+    Source:
+        KLM guide
+        Table 8.3.1.3.3.1-1. Format of packed LAC/HRPT Data Sets (Version 2, pre-April 28, 2005).
+        Table 8.3.1.3.3.2-1. Format of LAC/HRPT Data Record for NOAA-N (Version 5, post-November 14,
+                             2006, all spacecraft).
+        Table 8.3.1.4.3.1-1. Format of packed GAC Data Record for NOAA KLM (Version 2, pre-April 28, 2005).
+        Table 8.3.1.4.3.2-1. Format of GAC Data Record for NOAA-N (Version 4, post-January 25, 2006,
+                             all spacecraft).
+
+    Note:
+        Table 8.3.1.3.3.1-1. and Table 8.3.1.4.3.1-1. define bit: 21 as
+        "frame sync word not valid"
+        Table 8.3.1.3.3.2-1. and Table 8.3.1.4.3.2-1. define bit: 21 as
+        "flywheeling detected during this frame"
+    """
+    FATAL_FLAG = 2**0  # Data should not be used for product generation
+    TIME_ERROR = 2**1  # Time sequence error detected within this scan
+    DATA_GAP = 2**2  # Data gap precedes this scan
+    CALIBRATION = 2**3  # Insufficient data for calibration
+    NO_EARTH_LOCATION = 2**4  # Earth location data not available
+    CLOCK_UPDATE = 2**5  # First good time following a clock update (nominally 0)
+    INSTRUMENT_CHANGE = 2**6  # Instrument status changed with this scan
+    BIT_SYNC_STATUS = 2**7  # Sync lock dropped during this frame
+    SYNC_ERROR = 2**8  # Frame sync word error greater than zero
+    FRAME_SYNC_LOCK = 2**9  # Frame sync previously dropped lock
+    SYNC_INVALID = 2**10  # Frame sync word not valid
+    FLYWHEELING = 2**10  # Flywheeling detected during this frame
+    BIT_SLIPPAGE = 2**11  # Bit slippage detected during this frame
+    TIP_PARITY = 2**23  # TIP parity error detected
+    # Reflected Sunlight (RS) detected
+    CH_3B_RS = 2**24
+    CH_3B_RS_ANOMALY = 2**25
+    CH_4_RS = 2**26
+    CH_4_RS_ANOMALY = 2**27
+    CH_5_RS = 2**28
+    CH_5_RS_ANOMALY = 2**29
+    # using same name as for POD
+    CH_3_CONTAMINATION = 2**24  # Channel 3 solar blackbody contamination
+    CH_4_CONTAMINATION = 2**26  # Channel 4 solar blackbody contamination
+    CH_5_CONTAMINATION = 2**28  # Channel 5 solar blackbody contamination
```

Comment:
Something is fishy here, 2**24, 2**26 and 2**28 are used twice. Does this work?

Reply:
Yes, it is possible to define several aliases, which I did to have the same names as in the POD QFlags. In this case both flags have the same meaning so it should not be very confusing. A more "critical" flag is
```
    SYNC_INVALID = 2**10  # Frame sync word not valid
    FLYWHEELING = 2**10  # Flywheeling detected during this frame
```
see the docstring note.
I could implement the correct Flag in the constructor after checking the date, but the static implementation is simpler and I hope that someone using this flag knows which definition is applicable...
Any suggestion?

Reply:
Works for me. Just to make it clearer, maybe having the pod name close the klm would help?
```suggestion
    CH_3B_RS = 2**24
    # POD compatible name, Channel 3 solar blackbody contamination
    CH_3_CONTAMINATION = 2**24
    CH_3B_RS_ANOMALY = 2**25
    CH_4_RS = 2**26
    # POD compatible name, Channel 4 solar blackbody contamination
    CH_4_CONTAMINATION = 2**26
    CH_4_RS_ANOMALY = 2**27
    CH_5_RS = 2**28
    # POD compatible name, Channel 5 solar blackbody contamination
    CH_5_CONTAMINATION = 2**28
    CH_5_RS_ANOMALY = 2**29
```